src/pybloomfilter.pyx

```diff
@@ -98,23 +100,32 @@ cdef class BloomFilter:
             # when we choose to round down when the calculated optimal number of
             # hashes is fractional)."
 
-            assert(error_rate > 0.0 and error_rate < 1.0), "error_rate allowable range (0.0,1.0) %f" % (error_rate,)
-            num_hashes = max(int(math.floor(math.log(1.0 / error_rate, 2.0))),1)
+            if not (error_rate > 0.0 and error_rate < 1.0):
+                raise ValueError("error_rate allowable range (0.0, 1.0) %f" % (error_rate,))
+
+            if hash_seeds:
+                num_hashes = len(hash_seeds)
+            else:
+                num_hashes = max(int(math.floor(math.log(1.0 / error_rate, 2.0))),1)
```

Comment:
- `math.floor` returns an `int` -- casting this isn't required.
- `math.log2` would be more obvious here instead of passing `base` in args.

`num_hashes = max(math.floor(math.log2(1 / error_rate), 1)`

Reply:
Will do.

Reply:
7b5c24d